### PR TITLE
Adding alternate option for client authentication

### PR DESCRIPTION
**Pull request by jaypanchal**
User can pass 'clientId', 'clientSecret' and 'refreshToken' as an
authentication with
Set Of Authentication Method 1: 'serviceAccountId', 'keyFileLocation'

OR 
Set Of Authentication Method 2: 'clientId', 'clientSecret' and
'refreshToken'

### CODE REVIEW

**Comment by nightscape** [comment]
Thank for the PR! Some propositions:
You should be able to write the config reading more elegantly using monadic `Option` style like this:
```scala
val credentialsFromSecretAndToken = for {
  clientId <- parameters.get("clientId")
  clientSecret <- parameters.get("clientSecret")
  refreshToken <- parameters.get("refreshToken")
} yield {
  val credentials = new GoogleCredential.Builder().setTransport(httpTransport)
    .setJsonFactory(jsonFactory)
    .setClientSecrets(clientId, clientSecret)
    .build()
  credentials.setRefreshToken(refreshToken)
  credentials
}

val credentialsFromKeyFile = for {
  keyFileLocation <- parameters.get("keyFileLocation")
  serviceAccountId <- parameters.get("serviceAccountId")
} yield {
  new GoogleCredential.Builder().setTransport(httpTransport)
    .setJsonFactory(jsonFactory)
    .setServiceAccountId(serviceAccountId)
    .setServiceAccountPrivateKeyFromP12File(new File(keyFileLocation))
    .setServiceAccountScopes(util.Arrays.asList(AnalyticsScopes.ANALYTICS_READONLY))
    .build()
}

val credentials = credentialsFromSecretAndToken.orElse(credentialsFromKeyFile).getOrElse(throw new Exception("Incomplete credential information. Please provide valid credentials. See README file instructions for more info."))
```
Can you try using that (and fix remaining errors, can't test right now)?
Also please fix any style errors that Travis is complaining about :slightly_smiling_face: 

Thanks!

**Comment by jaypanchal** [comment]
Alright! Great, thanks for attention. I'll be working on it to make it... As this should be... @nightscape  Actually i am usually coding in java frameworks... so Scala is new one for me... :) Expect more suggestions. And i'll be definitely testing all this stuff.

**Comment by jaypanchal** [comment]
Do we have any independent or Eclipse plugin tool available to check Travis Style standards? Or Do we have to manually go through build logs?

**Comment by nightscape** [comment]
Just check the `.travis.yml` file, it contains all the command Travis executes and you should be able to run them manually.

**Comment by jaypanchal** [comment]
@nightscape Build is showing success status now. 
I have also updated README instructions file. Newer code is tested on localhost and running smooth. 
Anything else you would love to add up from your side?

File: README.md [hunk 1]
```diff
@@ -66,6 +71,24 @@ val df = sqlContext.read
...
 
+OR 
+
+Option 2 : Authentication with Client ID, Client Secret and Refresh Token'
+
+val sqlContext = new SQLContext(sc)
+val df = sqlContext.read
+    .format("com.crealytics.google.analytics")
+    .option("clientId", "XXXXXXXX-xyxyxxxxyxyxxxxxyyyx.apps.googleusercontent.com")
+	.option("clientSecret", "73xxYxyxy-XXXYZZx-xZ_Z")
```
Comment (nightscape) [inline README.md]: Indentation looks different from the rest. Tabs vs. spaces?

**Comment by nightscape** [comment]
Hey @jaypanchal, thanks for incorporating the and testing the proposals!
I've made one last comment about formatting in the README, once that is done I'll merge and deploy a new version.

**Comment by jaypanchal** [comment]
@nightscape I have updated README notes. But i don't know why i am getting those lines under brown lines. Would you please suggest me the reason ? or How do i correct it?

**Comment by nightscape** [comment]
Ah, just noticed that you put clear text into the 

> \```scala

code area. You need to split that code area into two and put the comments before the code areas.

**Comment by nightscape** [comment]
Hi @jaypanchal,

I made some mini-changes to the README (and apparently was able to directly commit to your fork :astonished:).
I'm happy with the PR now, so if you give me a final :+1: I'll merge and release it.

**Comment by jaypanchal** [comment]
@nightscape thanks man! :)
We can go ahead with merge and release.

**Comment by nightscape** [comment]
Ok, I just released `1.1.1`, it should appear in the repos within the next hours.

**Comment by jaypanchal** [comment]
@nightscape  Cool! :+1:

**Pull request merged**